lib/testers/rule-tester.js

```diff
@@ -490,7 +490,19 @@ RuleTester.prototype = {
                 }
 
                 if (item.hasOwnProperty("output")) {
-                    const fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+                    let fixResult = SourceCodeFixer.applyFixes(eslint.getSourceCode(), messages);
+
+                    if (item.multipass) {
```

Comment:
> When you say this would be a change to core, do you mean that RuleTester is a public API that might be used externally? I thought it was just an internal test util and I didn't realize there would be hesitation with changing it.

Yes, `RuleTester` is exported as a property of the `eslint` object so that plugin authors can use it to test custom rules.

I think the best solution is probably to change the five tests to only assert the result of the first pass. (I agree that there is probably room for improvement in the `RuleTester` API as far as multipass is concerned -- I just think that should be a separate issue from this bugfix so that it can be discussed thoroughly.